src/System.Console/src/System/ConsolePal.Unix.cs

```diff
@@ -1125,6 +1123,26 @@ internal void Unregister()
                 _handlerRegistered = false;
                 Interop.Sys.UnregisterForCtrl();
             }
+
+            private static void OnBreakEvent(Interop.Sys.CtrlCode ctrlCode)
+            {
+                // This is called on the native signal handling thread. We need to move to another thread so signal handling is
+                // not blocked. Otherwise we may get deadlocked when the handler depends on work triggered from the signal handling thread.
+                // We spin up a new thread so the break event is handled promptly.
```

Comment:
Could you augment the comment with something like:
"We use a new thread rather than queueing to the ThreadPool in case the ThreadPool is saturated, in order to prioritize handling the ctrlCode."